sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java

```diff
@@ -78,6 +92,231 @@
   private static final Logger LOG = LoggerFactory.getLogger(ApiSurface.class);
 
   /**
+   * {@link Matcher}s for use in {@link ApiSurface} related tests that aim to keep the public API
+   * conformant to a hard-coded policy by controlling what classes are allowed to be exposed by an
+   * API surface.
+   */
+  // based on previous code by @kennknowles and others.
+  public static class Matchers {
+
+    private static class ClassInPackage extends TypeSafeDiagnosingMatcher<Class<?>> {
+
+      private final String packageName;
+
+      private ClassInPackage(final String packageName) {
+        this.packageName = packageName;
```

Comment:
I agree, I think accepting `Package` in `ApiSurface.ofPackage(...)` like you mention in one of your comments would be far more useful.